### ISSUE TITLE
Variant value is missing in some configs on version 1.11.1

### ISSUE THREAD

**Issue by kevinguitar** [amplitude/experiment-android-client#42]
Hello, after we updated the sdk to the latest version 1.11.1, we started to receive null for some config variants, looks like the expected value is assigned to the `key` field instead.

For example:
```
Variant(value=null, payload=null, expKey=null, key=off, metadata={default=true})
```

**Note**: The issue is not reproducible on 1.10.1, we'll rollback for now.

**Comment by tyiuhc** [comment]
Hi @kevinguitar, many thanks for raising this issue. It seems that local evaluation is returning a default variant in this case. To better assist on this issue, it would be helpful if you could let us know:

1. What variant was expected instead
2. What configuration (e.g. user_id, device_id) was when evaluation was called
3. The link to your flag if possible

Please feel free to respond either on this thread, or via email to tim.yiu@amplitude.com.

**Comment by kevinguitar** [comment]
@tyiuhc  I've emailed all the details to you, thanks for helping!

**Comment by tyiuhc** [comment]
For those who are experiencing the same situation:

In pre-1.11 versions, the "fetch()" network call will get a response with only variants for flags where the user is included; in v1.11, if the user is not included in the flag, a default variant is returned.

In all versions, when "variant()" is called, if the user is not included in the flag, a default variant is returned. In pre-1.11, this is "Variant(value=null, payload=null, expKey=null)", in 1.11, this is "Variant(value=null, payload=null, expKey=null, key=off, metadata={default=true})".
